onnxruntime/test/optimizer/initializer_test.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+
+#include <cstdio>
+#include <fstream>
+#include <numeric>
+
+#include "gsl/gsl"
+
+#include "gtest/gtest.h"
+
+#include "core/common/common.h"
+#include "core/framework/endian_utils.h"
+#include "test/util/include/asserts.h"
+
+namespace onnxruntime {
+namespace test {
+namespace {
+class FileDeleter {
+ public:
+  FileDeleter() = default;
+  FileDeleter(const PathString& path) : path_{path} {}
+  FileDeleter(FileDeleter&& other) { *this = std::move(other); }
+  FileDeleter& operator=(FileDeleter&& other) {
+    CleanUp();
+    path_ = std::move(other.path_);
```

Comment:
This seems like it could be somewhat unexpected behavior. Do we need to support operator=?